pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1377,6 +1377,43 @@ def func(ser):
     tm.assert_frame_equal(res, expected)
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "int64",
+        "uint64",
+        "int32",
+        "uint32",
+        "float64",
+        "float32",
+        "Int64",
+        "UInt64",
+        "Int32",
+        "UInt32",
+        "Float64",
+        "Float32",
+    ],
+)
+def test_groupby_agg_precision(dtype):
+    # GH33234
+    df = DataFrame(
+        {
+            "key1": ["a"],
+            "key2": ["b"],
+            "key3": [
+                pd.array([1583715738627261039], dtype=dtype)[0]
+                if dtype not in ["int32", "uint32", "Int32", "UInt32"]
+                else pd.array([4294967295], dtype=dtype)[0]
+                if dtype in ["uint32", "UInt32"]
+                else pd.array([2147483647], dtype=dtype)[0]
+            ],
+        }
+    )
+    expected = df[["key3"]]
+    result = df.groupby(["key1", "key2"]).agg(lambda x: x).reset_index()[["key3"]]
```

Comment:
Could you construct `expected` such that `result` is just `df.groupby(["key1", "key2"]).agg(lambda x: x)`?

Reply:
Done